tests/flows/test_vesting.py

```diff
@@ -171,3 +172,143 @@ def test_main(
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet, cap=to_wei(1000)
+    )
+    mint_fake_OCEAN(config)
+    initial_ocean_liq = to_wei(200)
+    ocean_contract = ERC20Token(web3=web3, address=get_address_of_type(config, "Ocean"))
+    ocean_contract.approve(
+        get_address_of_type(config, "Router"), initial_ocean_liq, publisher_wallet
+    )
+    vested_blocks = factory_router.get_min_vesting_period()
+    percentages = [0.01, 0.1, 1]
+    checkpoint_blocks = list(map(lambda p: int((vested_blocks * p) / 100), percentages))
+
+    vesting_amount = initial_ocean_liq // 100 * 9
+
+    tx = erc20_token.deploy_pool(
+        rate=to_wei(1),
+        basetoken_decimals=ocean_contract.decimals(),
+        vesting_amount=vesting_amount,
+        vested_blocks=vested_blocks,
+        initial_liq=initial_ocean_liq,
+        lp_swap_fee=to_wei("0.003"),
+        market_swap_fee=to_wei("0.001"),
+        ss_contract=get_address_of_type(config, "Staking"),
+        basetoken_address=ocean_contract.address,
+        basetoken_sender=publisher_wallet.address,
+        publisher_address=publisher_wallet.address,
+        market_fee_collector=get_address_of_type(config, "OPFCommunityFeeCollector"),
+        pool_template_address=get_address_of_type(config, "poolTemplate"),
+        from_wallet=publisher_wallet,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    pool_event = erc20_token.get_event_log(
+        ERC721FactoryContract.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert pool_event[0].event == "NewPool", "Cannot find NewPool event"
+    assert pool_event[0].args.ssContract == get_address_of_type(config, "Staking")
+    bpool_address = pool_event[0].args.poolAddress
+    bpool = BPool(web3, bpool_address)
+    assert bpool.is_finalized() is True
+
+    side_staking = SideStaking(
+        web3=web3, address=get_address_of_type(config, "Staking")
+    )
+    vesting_created_event = side_staking.get_event_log(
+        SideStaking.EVENT_VESTING_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert vesting_created_event, "Cannot find event VestingCreated."
+    assert vesting_created_event[0].args.datatokenAddress == erc20_token.address
+    assert (
+        vesting_created_event[0].args.publisherAddress
+        == erc20_token.get_payment_collector()
+    )
+    assert (
+        vesting_created_event[0].args.vestingEndBlock
+        == vested_blocks + tx_receipt.blockNumber
+    )
+    assert vesting_created_event[0].args.totalVestingAmount == vesting_amount
+
+    # check if vesting is correct
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+    assert available_vesting_before == 0
+
+    # advance checkpoint_block blocks to see if available vesting increased
+    for checkpoint_block in checkpoint_blocks:
```

Comment:
I would suggest a single loop through all the blocks, and the following logic:
- send ether to increase block count
- if block is not a checkpoint block, continue
- (we are implicitly in the case where the block is a checkpoint one, so in the loop), do the assert.

How long do these tests take, btw? If it's not a very expensive test, time-wise, then there is no point in marking it as slow. Can you please benchmark it first, then let's decide whether it is `slow`-worthy or not :)